flopy/discretization/modeltime.py

```diff
@@ -7,61 +12,221 @@ class ModelTime:
 
     Parameters
     ----------
-    stress_periods : pandas dataframe
-        headings are: perlen, nstp, tsmult
-    temporal_reference : TemporalReference
-        contains start time and time units information
+    perlen : list, np.ndarray
+        list or numpy array of stress period lengths
+    nstp : list, np.ndarray
+        list or numpy array of number of time steps per stress period
+    tsmult : list, np.ndarray
+        list or numpy array of time-step mult infomation
+    time_units : int or str
+        string or pre-mf6 integer representation (ITMUNI) of time units
+    start_datetime : various objects
+        user-supplied starting datetime representation. Please see the
+        ModelTime.parse_datetime documentation for a list
+        of the supported representation types
+    steady_state : list, np.ndarray
+        optional list or numpy array of boolean flags that indicate if
+        stress periods are steady-state or transient
     """
 
     def __init__(
         self,
-        period_data=None,
-        time_units="days",
+        perlen,
+        nstp,
+        tsmult=None,
+        time_units=None,
         start_datetime=None,
         steady_state=None,
     ):
-        self._period_data = period_data
-        self._time_units = time_units
-        self._start_datetime = start_datetime
+        nrecs = len(perlen)
+        if tsmult is None:
+            tsmult = np.full((nrecs,), 1)
+
+        if nrecs != len(nstp):
+            raise AssertionError()
```

Comment:
should these be `ValueError`s that say what went wrong? For example, `"perlen and nstp inputs must have the same length"`